### ISSUE TITLE
[BUG] Teams ignores input.text field values after updating an adaptive card and does not send them back to the bot. 

### ISSUE THREAD

**Issue by joschi27** [MicrosoftDocs/msteams-docs#4863]
Good day,

TLDR; Adaptive cards in Teams don't properly check associatedInputs (required fields) when card got updated.

(This bug is different to the other one i just submitted!)

## Issue Description:
The issue is with updated adaptive cards. When an adaptive card gets updated by a bot, input fields that were already filled out don't get populated with data the next time the card is submitted.

I only tested this with the windows desktop client.
I have not tested this with the Input.Submit or any other types of actions.

## How to reproduce:
1. Send the JSON (at the end) as an adaptive card to a teams channel
2. Insert any input into the pinCode input field
3. Press Button "Confirm Positive".
4. In the bot backend, send back the same card. (in our case, the entered pincode is wrong and we insert a new textBlock with the separate error message)
5. Press the "Confirm Positive" Button again, without changing anything

### Expected Behaviour
The card should be sent and Action.Execute honored as the pinCode field is valid.

OR

After the bot sends the updated card, the input.text field should be reset. (which would fix the problem but be a bit more infuriating for the user)

### Actual Behaviour
The card is not sent and the errorMessage on the Input.Text is displayed, as if the input field were empty.
//edit - i just found out that teams actually thinks the input.text field is empty after the card gets updated in any case.

### Workaround
Don't use associatedInput feature, errorMessage property or isRequired property and validate everything on the server, updating the card with a custom error message text. - This creates unnecessary traffic though and is not the way adaptive cards should work.

Json to send: 
```
{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.4",
    "body": [
        {
            "type": "Container",
            "containerStyle": "attention",
            "spacing": "none",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "size": "large",
                                    "text": "Test Page",
                                    "wrap": true
                                },
                                {
                                    "type": "TextBlock",
                                    "spacing": "small",
                                    "isSubtle": true,
                                    "wrap": true,
                                    "text": "Triggered by the novaalert teams-bot"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "spacing": "medium",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "PIN"
                },
                {
                    "type": "Input.Text",
                    "placeholder": "enter pin code",
                    "id": "pinCode",
                    "isMasked": true,
                    "style": "password",
                    "isRequired": true,
                    "errorMessage": "Please enter the correct PIN."
                }
            ]
        }
    ],
    "actions": [
        {
            "type": "Action.Execute",
            "fallback": "Action.Submit",
            "title": "Confirm positive",
            "data": {
                "value": "respond_positive"
            },
            "style": "positive",
            "associatedInputs": "Auto"
        }
    ]
}
```

**Comment by sayali-MSFT** [comment]
@joschi27 - We are looking into this I will get back to you soon.

**Comment by sayali-MSFT** [comment]
@joschi27 - Could you please check on below documentation:-
1. https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-teams-catering/csharp
2. https://docs.microsoft.com/en-us/adaptive-cards/authoring-cards/universal-action-model#actionexecute

Let us know if you have any query.

**Comment by sayali-MSFT** [comment]
@joschi27 - Could you please confirm if your issue has resolved with above suggestion or still looking for any help?

**Comment by joschi27** [comment]
Hi,
I don't think this is an issue on our program's side nor is it a help request, I believe this is a bug in the way microsoft teams handles adaptive cards, and I would like to see if it really is a bug (if somebody else can reproduce it).

Maybe some additional info; we're generating adaptive cards on our azure bot without any data - see the example json payload.

Also, it may be connected to this issue (which seems to be a symptom of the bug we're experiencing) - https://github.com/MicrosoftDocs/msteams-docs/issues/4862

MS Teams seems to just clear the adaptive card data once any button (Action.Execute) is clicked. So when the user does not click into the textfield (which still has a text in it, but the underlying data is null) teams does either not send the card (because the isRequired property is present) or just sends null if the isrequired property is not present.

We have "fixed" this with a workaround, we do the work on the server side now - as i have already expressed in my ticket, that is not the way adaptive cards should work. 

Thanks.

**Comment by sayali-MSFT** [comment]
@joschi27 - We are able to repro your scenario.
Could you please provide the link where you raised any tickets or bug .
If not then you can raised user voice here
https://microsoftteams.uservoice.com/forums/555103-public/filters/new

**Comment by v-chetsh** [comment]
We are closing this issue for now, Please feel free to reopen it, if required.

**Comment by joschi27** [comment]
Sorry for the late reply - So, on future bugs that I find should I just raise a uservoice instead of writing a github issue?

**Comment by ChetanSharma-msft** [comment]
Hello @joschi27 - Hope you are doing well!!
For any bugs, you can raise it on GitHub only but if any new feature request, you can raise a UserVoice.

Thanks!!
